src/plugin.js

```diff
@@ -364,66 +364,69 @@ var zoomPlugin = {
 		if (!options.zoom || !options.zoom.enabled) {
 			return;
 		}
-		if (options.zoom.drag) {
-			chartInstance.zoom._mouseDownHandler = function(event) {
+
+		chartInstance.zoom._mouseDownHandler = function(event) {
+			if (chartInstance.options.zoom.drag) {
 				chartInstance.zoom._dragZoomStart = event;
-			};
-			node.addEventListener('mousedown', chartInstance.zoom._mouseDownHandler);
+			}
+		};
+		node.addEventListener('mousedown', chartInstance.zoom._mouseDownHandler);
 
-			chartInstance.zoom._mouseMoveHandler = function(event) {
-				if (chartInstance.zoom._dragZoomStart) {
-					chartInstance.zoom._dragZoomEnd = event;
-					chartInstance.update(0);
+		chartInstance.zoom._mouseMoveHandler = function(event) {
+			if (chartInstance.options.zoom.drag && chartInstance.zoom._dragZoomStart) {
+				chartInstance.zoom._dragZoomEnd = event;
+				chartInstance.update(0);
+			}
+		};
+		node.addEventListener('mousemove', chartInstance.zoom._mouseMoveHandler);
+
+		chartInstance.zoom._mouseUpHandler = function(event) {
+			if (chartInstance.options.zoom.drag && chartInstance.zoom._dragZoomStart) {
```

Comment:
I'd probably prefer to do the inverse to save from having to indent the rest of the code as much:

```
if (!chartInstance.options.zoom.drag || !chartInstance.zoom._dragZoomStart) {
  return;
}
```

Reply:
But probably only for a longer method like this one. For a method that's only a few lines, what you have now is shorter